src/server/scripts/Spells/spell_dh.cpp

```diff
@@ -231,6 +363,82 @@ struct areatrigger_dh_sigil_of_chains : AreaTriggerAI
     }
 };
 
+// 131347 - Glide
+class spell_dh_glide : public SpellScript
+{
+    PrepareSpellScript(spell_dh_glide);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DH_GLIDE_KNOCKBACK, SPELL_DH_GLIDE_DURATION });
+    }
+
+    SpellCastResult CheckCast()
+    {
+        Unit* caster = GetCaster();
+        if (!caster->IsFalling())
+            return SPELL_FAILED_NOT_ON_GROUND;
+
+        if (caster->IsMounted() || caster->GetVehicleBase())
+            return SPELL_FAILED_DONT_REPORT;
+
+        return SPELL_CAST_OK;
+    }
+
+    void HandleCast()
+    {
+        Player* caster = GetCaster()->ToPlayer();
+        if (!caster)
+            return;
+
+        caster->CastSpell(caster, SPELL_DH_GLIDE_KNOCKBACK, true);
+        caster->CastSpell(caster, SPELL_DH_GLIDE_DURATION, true);
+
+        auto history = caster->GetSpellHistory();
+        history->SendForcedCooldown(SPELL_DH_VENGEFUL_RETREAT_TRIGGER, 250);
+        history->SendForcedCooldown(SPELL_DH_FEL_RUSH, 250);
+    }
+
+    void Register() override
+    {
+        if (m_scriptSpellId == SPELL_DH_GLIDE)
+        {
+            OnCheckCast += SpellCheckCastFn(spell_dh_glide::CheckCast);
+            BeforeCast += SpellCastFn(spell_dh_glide::HandleCast);
+        }
+    }
+};
+
+// 131347 - Glide
+// 197154 - Glide
+class spell_dh_glide_AuraScript : public AuraScript
+{
+    PrepareAuraScript(spell_dh_glide_AuraScript);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DH_GLIDE, SPELL_DH_GLIDE_DURATION });
+    }
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetTarget()->RemoveAura(SPELL_DH_GLIDE);
+    }
+
+    void OnRemoveGlide(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        GetTarget()->RemoveAura(SPELL_DH_GLIDE_DURATION);
+    }
+
+    void Register() override
+    {
+        if (m_scriptSpellId == SPELL_DH_GLIDE_DURATION)
```

Comment:
These are two completely separate scripts, different `OnRemove*` functions, different params to `AuraEffectRemoveFn`, so why did you shove them both into the same class, especially when `spell_dh_glide` should apply to only one of them?

Reply:
alright alright.. i just wanted to save the extra code